src/client.js

```diff
@@ -457,11 +458,23 @@ Client.prototype.quit = function() {
 };
 
 Client.prototype.clientAttach = function(socketId) {
+	if (_.size(this.attachedClients) === 0) {
+		this.networks.forEach(function(network) {
+			network.irc.raw("AWAY");
+		});
+	}
+
 	this.attachedClients[socketId] = this.lastActiveChannel;
 };
 
 Client.prototype.clientDetach = function(socketId) {
 	delete this.attachedClients[socketId];
+
+	if (_.size(this.attachedClients) === 0) {
+		this.networks.forEach(function(network) {
+			network.irc.raw("AWAY", this.awayMessage);
```

Comment:
As I understand [MDN's article on arrow functions](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Arrow_functions#No_binding_of_this), this would come for free if `forEach` was using one of them. I haven't used them much, so I might be wrong.